modules/gateway-task/main.tf

```diff
@@ -202,3 +211,70 @@ resource "aws_ecs_task_definition" "this" {
     ],
   )
 }
+
+resource "aws_ecs_service" "this" {
+  name            = local.service_name
+  cluster         = var.ecs_cluster_arn
+  task_definition = aws_ecs_task_definition.this.arn
+  desired_count   = 1
+  network_configuration {
+    subnets          = var.subnets
+    security_groups  = var.security_groups
+    assign_public_ip = var.assign_public_ip
+  }
+  dynamic "load_balancer" {
+    for_each = local.load_balancer
+    content {
+      target_group_arn = load_balancer.value["target_group_arn"]
+      container_name   = load_balancer.value["container_name"]
+      container_port   = load_balancer.value["container_port"]
+    }
+  }
+  launch_type            = var.launch_type
+  propagate_tags         = "TASK_DEFINITION"
+  enable_execute_command = true
+}
+
+resource "aws_lb" "this" {
+  count              = var.lb_enabled ? 1 : 0
+  name               = local.service_name
+  internal           = false
+  load_balancer_type = "network"
+  subnets            = var.lb_subnets
+}
+
+resource "aws_lb_target_group" "this" {
+  count                = var.lb_enabled ? 1 : 0
+  name                 = local.service_name
+  port                 = tostring(local.wan_port)
+  protocol             = "TCP"
+  target_type          = "ip"
+  vpc_id               = var.lb_vpc_id
+  deregistration_delay = 120
+  health_check {
+    protocol = "TCP"
+  }
+}
+
+resource "aws_lb_listener" "this" {
+  count             = var.lb_enabled ? 1 : 0
+  load_balancer_arn = aws_lb.this[0].arn
+  port              = tostring(local.wan_port)
+  protocol          = "TCP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.this[0].arn
+  }
+}
+
+resource "aws_security_group_rule" "lb_ingress_rule" {
+  count             = var.lb_enabled ? length(var.security_groups) : 0
```

Comment:
I'm thinking we shouldn't modify the security groups they pass in, unless they request it. (Also, we'd only need to add the rule on one of the security groups. AWS takes applies the union/sum of the rules from all the security groups)

Some thoughts:

1. We support creating a security group for them. We would add `lb_create_security_group = true` (default true). If true, we create a group + rule and append it to `var.security_groups`.
1. Also, add `lb_modify_security_group = true` and `lb_modify_security_group_id = <group-id>`. Then we would attach a rule to the specific security group they specify.
1. ???

What do you think?

Reply:
I agree that we probably shouldn't just modify the security groups that are passed in. I really like the idea of doing 1.

Maybe we do 2. if there is enough time before the release?

Reply:
Agreed, I think we should change this to do (1).

We can skip (2). And just document that they need to add a security group rule to allow ingress to the task from the load balancer.

Reply:
I ended up doing both (1) and (2) because it was pretty straight forward.